engine/src/stack2.cpp

```diff
@@ -1573,7 +1573,7 @@ MCCard *MCStack::getchildbyname(MCNameRef p_name)
                 return cptr;
             cptr = cptr->next();
         }
-        while (cptr != cards);
+        while (cptr != cards && cptr != savecards);
```

Comment:
This isn't quite right - whether cptr should be compared against cards or savecards is determined by the condition above:
```
MCCard *cptr;
if (editing != NULL && savecards != NULL)
  cptr = savecards;
else
  cptr = cards;
```
I suggest:
MCCard *cptr, *cptr_sentinal;
if (editing != NULL && savecards != NULL)
{
  cptr_sentinal = cptr = savecards;
}
else
{
  cptr_sentinal = cptr = cards;
}
```
Then the conditions here and in the next loops are `cptr != cptr_sentinal`.